keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -340,47 +352,87 @@ def next(prompt, cache, index):
 
     def generate(
         self,
-        prompt,
-        max_length,
+        inputs,
+        max_length=None,
     ):
-        """Generate text.
+        """Generate text given prompt `inputs`.
 
-        This method generates text based on given `prompt`. Generation will
-        continue until `max_length` is met, and all tokens generated after
-        `end_token` will be truncated. The sampling strategy can be set in
-        the `compile` method.
+        This method generates text based on given `prompt`. The sampling used
```

Comment:
It looks like `prompt` has been migrated to `inputs`

Reply:
It felt weird to call something that could be a dataset or a tuples of tensors including a mask a `prompt`. WDYT?

Reply:
It's fine to use prompt as a term but backtick `prompt` usually means an variable name. Maybe say based on the prompts in `inputs`?

Reply:
Oh got it! Misunderstood the comment. Will just use `inputs` everywhere for now.